server/graphql-schema.js

```diff
@@ -810,6 +833,10 @@ const graphqlSchema = gql`
         """
         ats: [At]!
         """
+        Get AT Version information, given an AT ID and version string
+        """
+        atVersion(atId: ID!, atVersion: String!): AtVersion!
```

Comment:
See comments below in bold:

> I was looking at this for a little while, trying to put my finger on what seemed off about it, and then I realized that there are two concepts here that have the name "atVersion". First, atVersion is a string, which contains the version string for the AT. Second, AtVersion is a type with fields like availability.
> 
> What I've been thinking is something like this:
> 
> ```
> type AtVersion {
>   id: ID!
>   name: String!
>   releasedAt: Timestamp!
> }
> ```
> 
> Some thoughts on this:
> 
> * Adding an ID makes some sense since we are going to allow editing the AtVersions. Otherwise the API gets really awkward where you need to provide the old AtVersion name in order to provide a new AtVersion name.

**I agree with you on using IDs. I'm curious about how each AtVersion would be associated with an AT. Perhaps, just leaving that column in the table?**

> * We would want to use an ID anywhere we are currently using the atVersion string. This does mean probably migrating the existing data (!) and I realize this might get a little hairy but ... I do think ... it's doable.

**This seems like something we have to do.**

> * With an ID, it would make editing the AtVersion name way more intuitive. The type is currently a bit special and tricky and it would be great to make it look like the other types.

**+1**

> * There is already a custom "scalar" type specifically for timestamps so I put it in there.
> * The reason I suggested "releasedAt" is that all the timestamps currently in the schema end with the word "at" which is sorta neat I guess.

**Agreed. Let's replace the availability field with this.**

> * The other big thing I proposed there is making the releasedAt a required field. I feel like it should be required? Or not? What would be a reason to leave it blank?

**The only reason I left it optional was for the existing entries without that data.**